### DISCUSSION TITLE
Introduce fluent-style type declaration

### DISCUSSION THREAD

**Discussion by murtukov** [webonyx/graphql-php#797] [Ideas]
I just wanted to ask, wouldn't it be reasonable to make possible to declare types in fluent style?

For example this:
```php
Type::nonNull(Type::listOf(Type::nonNull($this->get('Human'))))
```
could also be written in fluent style:
```php
$this->get('Human')->nonNull()->list()->nonNull()
```

**Comment by spawnia** [comment]
I like how this follows the direction of the corresponding SDL definition:

```graphql
[           Human    !          ]      !
$this->get('Human')->nonNull()->list()->nonNull()
```

It also:
- should be much more quick to write, autocompletion would provide the correct available methods
- provides type safety, useful with static analysis tools, e.g. you can not chain `nonNull()` two times

The downsides include:
- Parity with `graphql-js`, have they adopted a similar syntax?
- Maintenance burden, more code and larger API surface

Are you willing to provide a PR for this?

**Comment by murtukov** [comment]
@spawnia Yes, I will provide a PR, but a little later. Towards the end of December I would say.

**Comment by murtukov** [comment]
@spawnia 

> Parity with graphql-js, have they adopted a similar syntax?

Can't say. I've never used `graphql-js`
